web_src/js/components/RepoBranchTagSelector.vue

```diff
@@ -246,7 +246,35 @@ const sfc = {
         event.preventDefault();
         this.menuVisible = false;
       }
-    }
+    },
+    handleTabSwitch(mode) {
+      this.createTag = mode === 'tags';
+      this.mode = mode;
+      this.focusSearchField();
+      this.fetchBranchesOrTags();
+    },
+    async fetchBranchesOrTags() {
+      if (!['branches', 'tags'].includes(this.mode) || this.isLoading) return;
+      // only fetch when branch/tag list has not been initialized
+      if (this.hasListInitialized[this.mode] ||
+      (this.mode === 'branches' && !this.showBranchesInDropdown) || (this.mode === 'tags' && this.noTag)) return;
+      this.isLoading = true;
```

Comment:
Need a `finally` block in case network error happens

Reply:
Added in https://github.com/go-gitea/gitea/pull/25719/files#diff-588b4b6c148c056a96c4cb1732bb5a12b12ee31ed2e2938eaf751b44e5f9fd64